### PR TITLE
Issues/2122 add azure environ var (partially resolves #2122)

### LINKED ISSUE #2122: There is no zone environment variable for GCE or Azure
AWS has the option to export TOIL_AWS_ZONE. There should be one for GCE and Azure as well.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/TOIL-69)
┆Issue Number: TOIL-69

### PR DESCRIPTION
**Pull request by bencvdb**
Adds an environment variable for the user to specify a zone in which instances will be spun up.

### CODE REVIEW

File: src/toil/provisioners/azure/azureProvisioner.py [hunk 1]
```diff
@@ -211,8 +213,7 @@ def _checkIfClusterExists(self):
         try:
             self.callPlaybook(self.playbook['check-cluster'], ansibleArgs, wait=True)
         except RuntimeError:
-            logger.info("The cluster could not be created. Try deleting the cluster if it already exits.")
-            raise
+            raise RuntimeError("The cluster could not be created. Try deleting the cluster if it already exits.")
```
Comment (jessebrennan) [inline src/toil/provisioners/azure/azureProvisioner.py]: I like this, but I wonder if there's any info from the old RuntimeError that is now lost.

File: src/toil/provisioners/azure/__init__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+def getAlternativeAzureZone(defaultZone):
+    '''
```
Comment (jessebrennan) [inline src/toil/provisioners/azure/__init__.py]: Here I go nitpicking :) """ instead of '''?

File: src/toil/provisioners/azure/__init__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    Find an appropriate azure zone.
+
+    Look for an environment variable or return a default as provided.
+
+    :param defaultZone: The zone specified in the leader metadata.
+    :return zone:  The zone.
+    '''
+
+    return os.environ.get('TOIL_AZURE_ZONE') or defaultZone
```
Comment (jessebrennan) [inline src/toil/provisioners/azure/__init__.py]: github complains about newline at end of file :/

File: src/toil/provisioners/azure/__init__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+def getAzureZone(defaultZone=None):
+    '''
```
Comment (jessebrennan) [inline src/toil/provisioners/azure/__init__.py]: Here I go nitpicking again :) do you think you can change these to """

File: src/toil/provisioners/azure/azureProvisioner.py (hunk 1 above)
Comment (bencvdb) [reply src/toil/provisioners/azure/azureProvisioner.py]: Oh from inside self.callPlaybook?

**Comment by DailyDreaming** [comment]
http://jenkins.cgcloud.info/job/toil-pull-requests/4219
```
src/toil/provisioners/azure/azureProvisioner.py:36: in <module>
    from toil.provisioners.azure import getAlternativeAzureZone
E   ImportError: cannot import name getAlternativeAzureZone
```

File: src/toil/provisioners/azure/azureProvisioner.py [hunk 5]
```diff
@@ -30,6 +30,7 @@
 from azure.common.credentials import ServicePrincipalCredentials
 from azure.mgmt.compute import ComputeManagementClient
 from azure.mgmt.network import NetworkManagementClient
+from toil.provisioners.azure import getAlternativeAzureZone
```
Comment (DailyDreaming) [inline src/toil/provisioners/azure/azureProvisioner.py]: I'm pretty sure you meant `from toil.provisioners.azure import getAzureZone` here.

File: docs/running/cloud/azure.rst [hunk 6]
```diff
@@ -105,3 +105,23 @@ After :ref:`prepareAzure` all you will need to do is specify the job store name
...
 
     $ python sort.py azure:<my-azure-account-name>:my-azure-jobstore
+
+Details about Launching a Cluster in Azure
+------------------------------------------
+
+Using the provisioner to launch a Toil leader instance is simple using the ``launch-cluster`` command. For example,
+to launch a cluster named "my-cluster" with a Standard_A2 leader in the westus zone, run ::
+
+    (venv) $ toil launch-cluster my-cluster --provisioner azure --leaderNodeType Standard_A2 --zone westus --keyPairName <your-AWS-key-pair-name>
```
Comment (DailyDreaming) [inline docs/running/cloud/azure.rst]: Still says `your-AWS-key-pair-name`.

File: docs/running/cloud/clusterUtils.rst [hunk 7]
```diff
@@ -204,12 +204,12 @@ exist yet, Toil will create it for you.
...
-  --zone ZONE           -z ZONE also accepted.  The AWS availability zone of the master. This
-                        parameter can also be set via the TOIL_AWS_ZONE
+  --zone ZONE           -z ZONE also accepted.  The availability zone of the leader. This
+                        parameter can also be set via the TOIL_AWS_ZONE or TOIL_AZURE_ZONE
                         environment variable, or by the ec2_region_name
-                        parameter in your .boto file, or derived from the
+                        parameter in your .boto file if using AWS, or derived from the
                         instance metadata if using this utility on an existing
-                        EC2 instance. Currently: us-west-1a
+                        EC2 instance. Currently: us-west-2a
```
Comment (DailyDreaming) [inline docs/running/cloud/clusterUtils.rst]: Good catch!

**Comment by DailyDreaming** [comment]
LGTM!

**Comment by DailyDreaming** [comment]
LGTM!

**Pull request merged**